launcher/src/main/java/org/apache/spark/launcher/SparkSubmitCommandBuilder.java

```diff
@@ -311,11 +311,15 @@ private List<String> buildSparkSubmitCommand(Map<String, String> env)
   }
 
   private void checkJavaOptions(String javaOptions) {
-    if (!isEmpty(javaOptions) && javaOptions.contains("Xmx")) {
-      String msg = String.format("Not allowed to specify max heap(Xmx) memory settings through " +
-        "java options (was %s). Use the corresponding --driver-memory or " +
-        "spark.driver.memory configuration instead.", javaOptions);
-      throw new IllegalArgumentException(msg);
+    if (!isEmpty(javaOptions)) {
+      for (String javaOption: CommandBuilderUtils.parseOptionString(javaOptions)) {
+        if (javaOption.startsWith("-Xmx")) {
+            String msg = String.format("Not allowed to specify max heap(Xmx) memory settings through " +
+                    "java options (was %s). Use the corresponding --driver-memory or " +
```

Comment:
indentation: should be 2 space as before

<img width="812" alt="image" src="https://github.com/apache/spark/assets/1475305/8ebf99d3-8db0-4b21-a167-93a9a8bd81e3">

Reply:
Done